### PR TITLE
DM-35186: Add focusZ to VisitInfo.

### PR DESCRIPTION
**Pull request by jbkalmbach**

### CODE REVIEW

**Review by timj — COMMENTED** [review]
I'm not really sure I know enough about our serialization policy to comment on this. @kfindeisen would you be able to look at these changes?

**Comment by kfindeisen** [comment]
Yes, I can look. Are you asking for me to become a reviewer? I noticed @timj already did the `obs_base` PR.

**Comment by timj** [comment]
@kfindeisen yes, if you could review the afw changes that would be great. I have indeed looked at the obs_base side.

File: src/image/VisitInfo.cc [hunk 1]
```diff
@@ -303,6 +309,7 @@ class VisitInfoFactory : public table::io::PersistableFactory {
         // Version-dependent fields
         std::string instrumentLabel = version >= 1 ? record.get(keys.instrumentLabel) : "";
         table::RecordId id = version >= 2 ? record.get(keys.idnum) : 0;
+        float focusZ = version >= 3 ? record.get(keys.focusZ) : nan;
```
Comment (kfindeisen) [inline src/image/VisitInfo.cc:312]:
```suggestion
        double focusZ = version >= 3 ? record.get(keys.focusZ) : nan;
```

File: src/image/VisitInfo.cc [hunk 2]
```diff
@@ -587,6 +598,7 @@ std::string VisitInfo::toString() const {
     buffer << "weather=" << getWeather() << ", ";
     buffer << "instrumentLabel='" << getInstrumentLabel() << "', ";
     buffer << "id=" << getId();
+    buffer << "focusZ=" << getFocusZ() << ", ";
```
Comment (kfindeisen) [inline src/image/VisitInfo.cc:601]: Comma should go after ID, not focus.

File: tests/test_visitInfo.py [hunk 3]
```diff
@@ -136,7 +139,8 @@ def computeLstHA(data):
                                   22.2*degrees, 33.3*degrees, 0.444),
                               weather=Weather(2.2, 3.3, 44.4),
                               instrumentLabel="TestCameraTwo",
-                              id=123456
+                              id=123456,
+                              focusZ=-1.5,
```
Comment (kfindeisen) [inline tests/test_visitInfo.py:143]: I'd suggest a number with a different absolute value. +1.5 and -1.5 might be hard to tell apart when debugging.

**Review by kfindeisen — APPROVED** [review]
Looks good,  just some typos and requested clarifications.

**Pull request merged**